cpu/fe310/clock.c

```diff
@@ -41,64 +39,67 @@ void clock_init(void)
+            PRCI_REG(PRCI_PLLDIV) = (PLL_FINAL_DIV_BY_1(1) | PLL_FINAL_DIV(0));
 
-    /* Configure PLL */
-    PRCI_REG(PRCI_PLLCFG) |= PLL_R(CONFIG_CLOCK_PLL_R) | PLL_F(CONFIG_CLOCK_PLL_F) | PLL_Q(CONFIG_CLOCK_PLL_Q);
+            /* Configure PLL */
+            PRCI_REG(PRCI_PLLCFG) |= PLL_R(CONFIG_CLOCK_PLL_R) | PLL_F(CONFIG_CLOCK_PLL_F) | PLL_Q(CONFIG_CLOCK_PLL_Q);
 
-    /* Disable PLL Bypass */
-    PRCI_REG(PRCI_PLLCFG) &= ~PLL_BYPASS(1);
+            /* Disable PLL Bypass */
+            PRCI_REG(PRCI_PLLCFG) &= ~PLL_BYPASS(1);
 
-    /* Now it is safe to check for PLL Lock */
-    while ((PRCI_REG(PRCI_PLLCFG) & PLL_LOCK(1)) == 0);
-#endif
+            /* Now it is safe to check for PLL Lock */
+            while ((PRCI_REG(PRCI_PLLCFG) & PLL_LOCK(1)) == 0);
+        }
 
-    /* Switch over to PLL Clock source */
-    PRCI_REG(PRCI_PLLCFG) |= PLL_SEL(PLL_SEL_PLL);
+        /* Switch over to PLL Clock source */
+        PRCI_REG(PRCI_PLLCFG) |= PLL_SEL(PLL_SEL_PLL);
 
-    /* Turn off the HFROSC */
-    PRCI_REG(PRCI_HFROSCCFG) &= ~ROSC_EN(1);
-#elif CONFIG_USE_CLOCK_HFROSC_PLL
-    PRCI_set_hfrosctrim_for_f_cpu(CONFIG_CLOCK_DESIRED_FREQUENCY, PRCI_FREQ_UNDERSHOOT);
-#else /* Clock HFROSC */
-    /* Disable Bypass */
-    PRCI_REG(PRCI_PLLCFG) &= ~PLL_BYPASS(1);
+        /* Turn off the HFROSC */
+        PRCI_REG(PRCI_HFROSCCFG) &= ~ROSC_EN(1);
+    }
+    else if (CONFIG_USE_CLOCK_HFROSC_PLL) {
+        PRCI_set_hfrosctrim_for_f_cpu(CONFIG_CLOCK_DESIRED_FREQUENCY, PRCI_FREQ_UNDERSHOOT);
+    }
+    else { /* Clock HFROSC */
+        /* Disable Bypass */
+        PRCI_REG(PRCI_PLLCFG) &= ~PLL_BYPASS(1);
 
-    /* Configure trim and divider values of HFROSC */
-    PRCI_REG(PRCI_HFROSCCFG) = (ROSC_DIV(CONFIG_CLOCK_HFROSC_DIV) | ROSC_TRIM(CONFIG_CLOCK_HFROSC_TRIM) | ROSC_EN(1));
+        /* Configure trim and divider values of HFROSC */
+        PRCI_REG(PRCI_HFROSCCFG) = (ROSC_DIV(CONFIG_CLOCK_HFROSC_DIV) | ROSC_TRIM(CONFIG_CLOCK_HFROSC_TRIM) | ROSC_EN(1));
 
-    /* Wait for HFROSC to be ready */
-    while ((PRCI_REG(PRCI_HFROSCCFG) & ROSC_RDY(1)) == 0);
+        /* Wait for HFROSC to be ready */
+        while ((PRCI_REG(PRCI_HFROSCCFG) & ROSC_RDY(1)) == 0);
 
-    /* Don't use PLL clock source */
-    PRCI_REG(PRCI_PLLCFG) &= ~PLL_SEL(PLL_SEL_PLL);
-#endif
+        /* Don't use PLL clock source */
+        PRCI_REG(PRCI_PLLCFG) &= ~PLL_SEL(PLL_SEL_PLL);
+    }
 }
 
 uint32_t cpu_freq(void)
 {
-#if CONFIG_USE_CLOCK_HFXOSC || CONFIG_USE_CLOCK_HFXOSC_PLL
-    return CLOCK_CORECLOCK;
-#else /* Clock frequency with HFROSC cannot be determined precisely from
-         settings */
-    /* If not done already, estimate the CPU frequency */
-    if (_cpu_frequency == 0) {
-        /* Ignore the first run (for icache reasons) */
-        _cpu_frequency = PRCI_measure_mcycle_freq(3000, RTC_FREQ);
-        _cpu_frequency = PRCI_measure_mcycle_freq(3000, RTC_FREQ);
+    if (CONFIG_USE_CLOCK_HFROSC || CONFIG_USE_CLOCK_HFROSC_PLL) {
```

Comment:
I would suggest to use `IS_ACTIVE` on all these checks as well. On the current state it does not hurt and when switching to Kconfig a boolean set to `n` is a not-defined macro.

Reply:
I think I had build failures when using this macro.

Reply:
Errors like:

```c
In file included from /work/riot/RIOT/cpu/fe310/clock.c:20:
/work/riot/RIOT/core/include/kernel_defines.h:195:52: error: pasting "__PREFIX_WHEN_" and "(" does not give a valid preprocessing token
 #define __is_active(val)              ___is_active(__PREFIX_WHEN_##val)
                                                    ^~~~~~~~~~~~~~
/work/riot/RIOT/core/include/kernel_defines.h:176:26: note: in expansion of macro '__is_active'
 #define IS_ACTIVE(macro) __is_active(macro)

```

Reply:
That's because of the usage of `(` in the macro values

Reply:
Ok, so sounds more like an issue with the macro itself :)

Reply:
Well, it's not really an issue with the macro, it's documented that it must be used without `(` http://doc.riot-os.org/kernel__defines_8h.html#a25e432312ddc824e161578831972028c The implementation is taken as is from the kernel. The reason it's meant to work like that is, that's the way CFLAGS are added when defined `CFLAGS=-DCONFIG_USE_CLOCK_HFROSC` and Kconfig generates the values. I think we could define the macros without `(`, as we are only using them for this kind of switches.

IMHO if there is a good moment to change this is now, so we don't have to come back at the end of the Kconfig migration to adapt all these lines.

Reply:
Ok, I made that change for both boards and made use of IS_ACTIVE where possible. I tested an it still works.